docs/current/742989-cookbook.md

```diff
@@ -169,6 +169,106 @@ The following code listing obtains a reference to the host working directory con
 
 [Learn more](./guides/421437-work-with-host-filesystem.md)
 
+### Add Git repository as directory to container
+
+The following code listing adds a remote Git repository branch to a container as a directory at the `/src` container path and then executes a command in the container to list the directory contents.
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+```go file=./cookbook/snippets/filesystem-operations/add-git-dir/main.go
+```
+
+</TabItem>
+<TabItem value="Node.js">
+
+```javascript file=./cookbook/snippets/filesystem-operations/add-git-dir/index.mjs
+```
+
+</TabItem>
+<TabItem value="Python">
+
+```python file=./cookbook/snippets/filesystem-operations/add-git-dir/main.py
+```
+
+</TabItem>
+</Tabs>
+
+### Add Git repository as directory to container with exclusions
+
+The following code listing adds a remote Git repository branch as a directory at the `/src` container path, excluding `*.md` files.
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+```go file=./cookbook/snippets/filesystem-operations/add-git-dir-exclude/main.go
+```
+
+</TabItem>
+<TabItem value="Node.js">
+
+```javascript file=./cookbook/snippets/filesystem-operations/add-git-dir-exclude/index.mjs
+```
+
+</TabItem>
+<TabItem value="Python">
+
+```python file=./cookbook/snippets/filesystem-operations/add-git-dir-exclude/main.py
+```
+
+</TabItem>
+</Tabs>
+
+### Add Git repository as directory to container with inclusions
```

Comment:
Ok "inclusions" also sounds weird but I'm having a hard time even coming up with an alternative. :smile: 

"Include" is kind of a weird name to work with because if you use it on its own it actually _excludes_ more than it includes. For me it's just easier to think of these two at the same time: first we include files (or include everything by default), then we exclude them.

What do you think of just getting rid of the separate include/exclude examples and reducing it to "with glob filters" or something? I think most people are already familiar with the term 'glob' in this context and might even try searching for it. But "with filters" is probably fine too.

Maybe I'm missing the point of cookbooks (have a lot of'em) but it just seems easier to explain these two in the same breath.

Reply:
Modified to use "with filters", combined some sections as a result.